presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java

```diff
@@ -320,46 +320,22 @@ else if (literalCalculationParameters.contains(parameterName)) {
     @JsonValue
     public String toString()
     {
-        // TODO: remove these hacks
-        if (base.equalsIgnoreCase(StandardTypes.ROW)) {
-            return rowToString();
-        }
-        else if (base.equalsIgnoreCase(StandardTypes.VARCHAR) &&
+        if (base.equalsIgnoreCase(StandardTypes.VARCHAR) &&
                 (parameters.size() == 1) &&
                 parameters.get(0).isLongLiteral() &&
                 parameters.get(0).getLongLiteral() == VarcharType.UNBOUNDED_LENGTH) {
             return base;
         }
-        else {
-            StringBuilder typeName = new StringBuilder(base);
-            if (!parameters.isEmpty()) {
-                typeName.append("(");
-                boolean first = true;
-                for (TypeSignatureParameter parameter : parameters) {
-                    if (!first) {
-                        typeName.append(",");
-                    }
-                    first = false;
-                    typeName.append(parameter.toString());
-                }
-                typeName.append(")");
-            }
-            return typeName.toString();
-        }
-    }
-
-    @Deprecated
-    private String rowToString()
-    {
-        verify(parameters.stream().allMatch(parameter -> parameter.getKind() == ParameterKind.NAMED_TYPE),
-                format("Incorrect parameters for row type %s", parameters));
 
-        String fields = parameters.stream()
-                .map(TypeSignatureParameter::getNamedTypeSignature)
-                .map(NamedTypeSignature::toString)
-                .collect(Collectors.joining(","));
-
-        return format("row(%s)", fields);
+        StringBuilder typeName = new StringBuilder(base);
+        if (!parameters.isEmpty()) {
+            typeName.append("(")
+                    .append(parameters.stream()
+                            .map(TypeSignatureParameter::toString)
+                            .collect(joining(",")))
+                    .append(")");
```

Comment:
There original author didn't leave a comment on why he chose to implement the logic here without using `joining`. I can come up with a reason though. Using joining can make this function have `O(N^2)` time complexity in the case of `array(array(...(array(bigint))...))`. I recommend keeping that logic as is.

I agree with removing the `rowToString` function here.